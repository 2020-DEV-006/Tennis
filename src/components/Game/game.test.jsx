import React from "react";
import ReactDOM from "react-dom";
import Game from "./Game";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("Game", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Game />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("render Game component correctly", () => {
    const GameComponent = renderer.create(<Game />).toJSON();
    expect(GameComponent).toMatchSnapshot();
  });
});
