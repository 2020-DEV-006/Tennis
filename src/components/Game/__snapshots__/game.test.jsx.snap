// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game render Game component correctly 1`] = `
<div
  className="tennisBackground"
>
  <h2
    className="tennisHeading"
  >
    Play Tennis
  </h2>
  <button
    className="resetGame"
    onClick={[Function]}
  >
    Reset Game
  </button>
  <div
    className="scoreBoard"
  >
    <h3>
      scoreBoard
    </h3>
    <div
      className="playerStyles"
    >
      Player1: 
      <div
        className="scoreStyles"
      >
        0
      </div>
    </div>
    <div
      className="playerStyles"
    >
      Player2: 
      <div
        className="scoreStyles"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="playerBox"
  >
    <div>
      <div
        className="playerHeading"
      >
        Player 1
      </div>
      <div
        className="player"
        defaultValue={1}
        onClick={[Function]}
        value={1}
      />
    </div>
    <div>
      <div
        className="playerHeading"
      >
        Player 2
      </div>
      <div
        className="player"
        defaultValue={2}
        onClick={[Function]}
        value={2}
      />
    </div>
  </div>
  <span />
  <span />
  0
</div>
`;

exports[`Game resets the game 1`] = `
<div
  className="scoreBoard"
>
  <h3>
    scoreBoard
  </h3>
  <div
    className="playerStyles"
  >
    Player1: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
  <div
    className="playerStyles"
  >
    Player2: 
    <div
      className="scoreStyles"
    >
      0
    </div>
  </div>
</div>
`;
